reddit.py

```diff
@@ -0,0 +1,112 @@
+"""Reddit source code and datastore model classes.
+
+"""
+import logging
+
+from granary import reddit as gr_reddit
+from granary import source as gr_source
+from oauth_dropins import reddit as oauth_reddit
+from oauth_dropins.webutil.util import json_dumps, json_loads
+import webapp2
+from webob import exc
+
+import models
+import util
+
+
+class Reddit(models.Source):
+  """A Reddit account.
+
+  The key name is the username.
+  """
+  GR_CLASS = gr_reddit.Reddit
+  OAUTH_START_HANDLER = oauth_reddit.StartHandler
+  SHORT_NAME = 'reddit'
+  TYPE_LABELS = {
+    'post': 'submission',
+    'comment': 'comment',
+  }
+
```

Comment:
to disable publish, i think all you need to do is add `CAN_PUBLISH = False` here.

Reply:
ah cool, i missed that.